lib/jazzy/jazzy_markdown.rb

```diff
@@ -13,10 +13,9 @@ def header(text, header_level)
                       .sub(/^-/, '')
                       .sub(/-$/, '')
 
-      "<a href='##{text_slug}' class='anchor' aria-hidden=true>" \
-        '<span class="header-anchor"></span>' \
```

Comment:
It might be tough to do exactly that because the anchor + span do not overlap the header.  You're right though that there could be a user theme that gives this anchor some visual or relies on this html somehow.  I understand if that rules out this approach.